### PR TITLE
Issue 3 user can edit entry

### PR DESCRIPTION
**Pull request by humbertovnavarro**

### CODE REVIEW

**Comment by humbertovnavarro** [comment]
image
image
image
image

**Comment by humbertovnavarro** [comment]
image
image

image

**Comment by humbertovnavarro** [comment]
Mobile
image
image
image
image
image

File: js/main.js [hunk 1]
```diff
@@ -30,12 +47,17 @@ function handleFormSubmit(event) {
     notes: $form.notes.value,
     entryID: data.nextEntryId
   };
-  data.entries.push(entry);
+  if(data.editing != null) {
+    data.entries[data.editing] = entry;
+    data.editing = null;
```
Comment (codingcodymiller) [inline js/main.js]: You shouldn't be giving a new id to an edited entry, only new entries get new ids, this should reuse the old id

File: js/main.js [hunk 2]
```diff
@@ -30,12 +50,17 @@ function handleFormSubmit(event) {
...
+  } else {
+    data.nextEntryId++;
+    data.entries.push(entry);
+  }
```
Comment (codingcodymiller) [inline js/main.js]: I don't see anything in your submit function that is generating a DOM tree for the new entry, does the new entry show on the entries list when you are navigated there without needing to refresh the page?

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: js/main.js (hunk 2 above)
Comment (humbertovnavarro) [reply js/main.js]: Yes.

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**